dali/core/tensor_shape_test.cc

```diff
@@ -1179,5 +1179,97 @@ TEST(TensorTest, WontCompile) {
   // TensorView<EmptyBackendTag, int, DynamicDimensions>{TensorView<EmptyBackendTag, int8_t, 4>{}};
 }
 
+
+TEST(AppendTest, NonuniformShapes) {
+  TensorListShape<4> tls0 = {{{1, 2, 3, 4}, {5, 6, 7, 8}}};
+  TensorListShape<-1> tls1 = {{{1, 2, 3, 4}, {5, 6, 7, 8}}};
+  TensorListShape<-1> tls2 = {{{9, 10, 11}, {13, 14, 15}}};
+  TensorListShape<3> tls3 = {{{17, 18, 19}, {22, 23, 24}}};
+  TensorListShape<4> tls4 = {{{17, 18, 19, 20}, {21, 22, 23, 24}}};
+
+  EXPECT_THROW(tls1.append({tls3, tls4}), std::invalid_argument);
+  EXPECT_THROW(tls1.append({tls1, tls3}), std::invalid_argument);
+  EXPECT_THROW(tls1.append({tls1, tls2}), std::invalid_argument);
+}
+
+
+TEST(AppendTest, ToEmpty) {
```

Comment:
```suggestion
TEST(AppendTest, ToEmpty) {
```
What does "emtpy" mean here?
Maybe a better name would be "Append"? Same with the variables below "empty_st" and "empty_dyn", the "empty" in their name is confusing.